src/Features/Core/Portable/CodeAnalysisSuggestions/CodeAnalysisSuggestionsCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,108 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.Copilot;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.PooledObjects;
+
+namespace Microsoft.CodeAnalysis.CodeAnalysisSuggestions
+{
+    [ExportCodeRefactoringProvider(LanguageNames.CSharp), Shared]
+    internal sealed class CodeAnalysisSuggestionsCodeRefactoringProvider
+        : CodeRefactoringProvider
+    {
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public CodeAnalysisSuggestionsCodeRefactoringProvider()
+        {
+        }
+
+        public override async Task ComputeRefactoringsAsync(CodeRefactoringContext context)
+        {
+            var (document, span, cancellationToken) = context;
+            var configService = document.Project.Solution.Services.GetRequiredService<ICopilotConfigService>();
+            var promptParts = await configService.TryGetCopilotConfigPromptAsync(CopilotConfigFeatures.CodeAnalysisSuggestions, document.Project, cancellationToken).ConfigureAwait(false);
+            if (!promptParts.HasValue || promptParts.Value.IsEmpty)
```

Comment:
strange to have to check .HasValue and .IsEmpty.  I assume this is an ImmutableArray.  In which case, do .IsDefaultOrEmpty